### PR TITLE
fix(windows): Avoid double start during setup

### LINKED ISSUE #4724: bug(windows): setup can cause an error in keymanx64 when it continues after Windows restart
**Describe the bug**

If Keyman Setup needs to restart Windows to complete setup, Keyman can startup (with its 'start with Windows' setting) while Setup is doing its final tasks. This can cause keymanx64 to try to start twice or similar.

This was discovered while testing #4649.

image

---

**Keyman for Windows/macOS/Linux:**
 - OS: Windows 10 20H2 
 - Keyman version: 14.0.264-beta

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #4724.

### CODE REVIEW

**Comment by mcdurdin** [comment]
# User Testing

Please try installing this update to see if it causes the error described in #4724.

@MakaraSok

File: windows/src/global/delphi/general/Keyman.System.UpgradeRegistryKeys.pas [hunk 1]
```diff
@@ -94,6 +94,8 @@ interface
   SRegKey_UpgradeBackupPath_LM = '\'+SRegKey_KeymanEngine_LM+'\Upgrade Backup\';  // I2642
   SRegKey_UpgradeBackupPath_CU = '\'+SRegKey_KeymanEngine_CU+'\Upgrade Backup\';  // I2642
 
+  SRegValue_UpgradeRunKeyman = 'Upgrade - Start Keyman with Windows';
```
Comment (darcywong00) [inline windows/src/global/delphi/general/Keyman.System.UpgradeRegistryKeys.pas:97]: Is this key for upgrading to Keyman 14.0+? Would it use something like `SRegKey_KeymanEngine140_??`

Comment (mcdurdin) [reply windows/src/global/delphi/general/Keyman.System.UpgradeRegistryKeys.pas:97]: It's just a temporary key during upgrade, which is why it's in UpgradeRegistryKeys.pas.

**Review by darcywong00 — APPROVED** [review]
lgtm.

There sure are a lot of registry keys under the hood :)

**Comment by mcdurdin** [comment]
> There sure are a lot of registry keys under the hood :)

Squirmy 🐍 feeling?

**Comment by MakaraSok** [comment]
Tested using the artifact of this build on Windows 10 Pro 20H2 x64. Upgrading from 11 stable to this test build.

Confirm that no error messages mentioned here https://github.com/keymanapp/keyman/issues/4724#issue-835502013 pops up after the upgrade was completed.

Upgrading from 12 stable is pending...

**Comment by MakaraSok** [comment]
_The test above is not meant for this PR; please ignore it._ 

**This test should have been for 14.0.262-beta upgraded to 14.0.266-beta-test-4726.**

**Now confirm that no error is experienced after the installation is completed. Keyman is running as usual.**

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.266-beta](https://keyman.com/downloads/releases/beta/14.0.266)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.20-alpha](https://keyman.com/downloads/releases/alpha/15.0.20)

**Pull request merged**